### PR TITLE
[REF-5806] S3 file source component

### PR DESCRIPTION
**Pull request by PMjoydas**
S3 file source component has been tested

### CODE REVIEW

File: components/Python/fileConnectors/s3_file_source/component.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  "description": "Import a bucket/key from S3 to a file",
+  "program": "main.py",
+  "componentClass": "S3FileSource",
+  "group": "Connectors",
+  "useMLOps": true,
+  "inputInfo": [],
+  "outputInfo": [
+    {
+        "description": "File name",
+        "label": "File-Name",
```
Comment (yakov-g) [inline components/Python/fileConnectors/s3_file_source/component.json]: File-Name  with capital letters and dash look very strange :)

File: components/Python/fileConnectors/s3_file_source/component.json [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        "defaultComponent": "",
+        "type": "str",
+        "group": "data"
+    }
+  ],
+  "arguments": [
+    {
+      "key": "aws_access_key_id",
+      "label": "AWS Access Key",
```
Comment (yakov-g) [inline components/Python/fileConnectors/s3_file_source/component.json]: again capital letters ... and further

File: components/Python/fileConnectors/s3_file_source/main.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        client.download_file(self._params["bucket"], self._params["key"], file_path)
+        fetch_elapsed_time = time.time() - fetch_start_time
+        if self._params["get_fetch_time"]:
+            mlops.set_stat("s3.inputFetchTimemsec", fetch_elapsed_time)
+
+        # get line-count for the file (loads file in memory)
+        # should help keep prediction latency NOT be IO-bound
+        if self._params["get_line_count"]:
+            line_count = len(open(file_path).readlines())
+            if not self._params["get_rand_n_samples"]:
```
Comment (yakov-g) [inline components/Python/fileConnectors/s3_file_source/main.py]: what is "get_rand_n_samples"? I haven't seen this in the rest of the patch.

**Review by yakov-g — APPROVED** [review]
Just fix and/or comment.

File: components/Python/fileConnectors/s3_file_source/component.json (hunk 1 above)
Comment (PMjoydas) [reply components/Python/fileConnectors/s3_file_source/component.json]: replaced it to "filename"

File: components/Python/fileConnectors/s3_file_source/main.py (hunk 3 above)
Comment (PMjoydas) [reply components/Python/fileConnectors/s3_file_source/main.py]: (actually it was for the demo) have got rid of the piece of code, this was a benign lingerance of the artifact .. have removed it.

File: components/Python/fileConnectors/s3_file_source/component.json (hunk 2 above)
Comment (PMjoydas) [reply components/Python/fileConnectors/s3_file_source/component.json]: Thanks for bring this up..
Sticking to capitalization of words of significance and lowercase for all others

**Review by sriram109 — APPROVED** [review]

**Pull request merged**